scenes/test_string.txt

```diff
@@ -174,57 +174,57 @@
 *gosub_scene cslib_string extract_tokens str_buffer "|" "tokens"
 *comment There are actually 6 values, but only 5 are used.
 *comment The user may check for this condition.
-*gosub test_finish 6 cslib_ret
-*gosub test_finish "A" tokens_1
-*gosub test_finish "B" tokens_2
-*gosub test_finish "C" tokens_3
-*gosub test_finish "D" tokens_4
-*gosub test_finish "E" tokens_5
-*gosub test_finish 5 tokens_max
+*gosub test_assert_equal 6 cslib_ret
+*gosub test_assert_equal "A" tokens_1
+*gosub test_assert_equal "B" tokens_2
+*gosub test_assert_equal "C" tokens_3
+*gosub test_assert_equal "D" tokens_4
+*gosub test_assert_equal "E" tokens_5
+*gosub test_assert_equal 5 tokens_max
 *gosub test_finish 5 tokens_count
 
-*comment Empty source.
+*comment Empty source. The array is unchanged.
 *gosub test_start "EXTRACT_TOKENS #4"
 *set str_buffer ""
 *gosub_scene cslib_string extract_tokens str_buffer "|" "tokens"
-*gosub test_finish 0 cslib_ret
-*gosub test_finish "" tokens_1
-*gosub test_finish "" tokens_2
-*gosub test_finish "" tokens_3
-*gosub test_finish "" tokens_4
-*gosub test_finish "" tokens_5
-*gosub test_finish 5 tokens_max
+*gosub test_assert_equal 0 cslib_ret
+*gosub test_assert_equal "A" tokens_1
+*gosub test_assert_equal "B" tokens_2
+*gosub test_assert_equal "C" tokens_3
+*gosub test_assert_equal "D" tokens_4
+*gosub test_assert_equal "E" tokens_5
+*gosub test_assert_equal 5 tokens_max
 *gosub test_finish 0 tokens_count
 
 *comment Only one token.
 *gosub test_start "EXTRACT_TOKENS #5"
 *set str_buffer "HI"
 *gosub_scene cslib_string extract_tokens str_buffer "|" "tokens"
-*gosub test_finish 1 cslib_ret
-*gosub test_finish "HI" tokens_1
-*gosub test_finish "" tokens_2
-*gosub test_finish "" tokens_3
-*gosub test_finish "" tokens_4
-*gosub test_finish "" tokens_5
-*gosub test_finish 5 tokens_max
+*gosub test_assert_equal 1 cslib_ret
+*gosub test_assert_equal "HI" tokens_1
+*gosub test_assert_equal "" tokens_2
+*gosub test_assert_equal "" tokens_3
+*gosub test_assert_equal "" tokens_4
+*gosub test_assert_equal "" tokens_5
+*gosub test_assert_equal 5 tokens_max
```

Comment:
_count

Reply:
I'm checking both, count is the next line. Better safe than sorry.